src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloudMultipleLabelsIT.java

```diff
@@ -0,0 +1,116 @@
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.instanceConfiguration;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.logs;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import com.google.jenkins.plugins.computeengine.ComputeEngineCloud;
+import com.google.jenkins.plugins.computeengine.client.ComputeClient;
+import hudson.model.Node;
+import hudson.model.labels.LabelAtom;
+import hudson.slaves.NodeProvisioner.PlannedNode;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+import java.util.logging.StreamHandler;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class ComputeEngineCloudMultipleLabelsIT {
+  private static Logger log = Logger.getLogger(ComputeEngineCloudMultipleLabelsIT.class.getName());
+
+  private static final String MULTIPLE_LABEL = "integration test";
+
+  @ClassRule public static Timeout timeout = new Timeout(5, TimeUnit.MINUTES);
+  @ClassRule public static JenkinsRule jenkinsRule = new JenkinsRule();
+
+  private static ByteArrayOutputStream logOutput = new ByteArrayOutputStream();
+  private static StreamHandler streamHandler = new StreamHandler(logOutput, new SimpleFormatter());
+  private static ComputeClient client;
+  private static Map<String, String> label = getLabel(ComputeEngineCloudMultipleLabelsIT.class);
+  private static Collection<PlannedNode> planned;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    log.info("init");
+    initCredentials(jenkinsRule);
+    ComputeEngineCloud cloud = initCloud(jenkinsRule);
+    handleClassLogs(streamHandler, ComputeEngineCloud.class.getName());
+    client = initClient(jenkinsRule, label, log);
+    handleClassLogs(streamHandler, ComputeClient.class.getName());
+
+    cloud.addConfiguration(
+        instanceConfiguration(
+            DEB_JAVA_STARTUP_SCRIPT,
+            NUM_EXECUTORS,
+            MULTIPLE_LABEL,
+            label,
+            false,
+            false,
+            NULL_TEMPLATE));
+    // Add a new node
+    planned = cloud.provision(new LabelAtom(LABEL), 1);
+    planned.iterator().next().future.get();
+  }
+
+  @AfterClass
+  public static void teardown() throws IOException {
+    ITUtil.teardown(streamHandler, logOutput, client, label, log);
+  }
+
+  @Test
+  public void testMultipleLabelsForJob() {
+    // For a configuration with multiple labels, test if job label matches one of the
+    // configuration's labels
+
+    // There should be a planned node
+    assertEquals(logs(streamHandler, logOutput), 1, planned.size());
```

Comment:
The name of this test doesn't seem to correspond to this assertion. I would also expect assertions testing for the expected labels.

Reply:
The naming in this class comes from the original two tests which this class replaces. Both had the same setup that is now in init. I then removed that common setup and left what remained here. The expected labels are tested in the other test method but that isn't clear from the naming either. I'll rework this.